### PR TITLE
Fix units, simplify code, add lasercut style

### PR DESCRIPTION
**Pull request by Neon22**
quite a major refactor. Sorry its not in many smaller pull requests.
I've changed style of center marks by adding gap variable.

### CODE REVIEW

File: EurorackPanelDesigner/EurorackPanelDesigner.inx [hunk 1]
```diff
@@ -5,13 +5,20 @@
...
   <dependency type="executable" location="extensions">inkex.py</dependency>
   <dependency type="executable" location="extensions">EurorackPanelDesigner.py</dependency>
+  <dependency type="executable" location="extensions">simplestyle.py</dependency>
   
+  <param name="desc_unit" type="description">Eurorack Panel maker:</param>
+  <param name="desc_unit" type="description">Width is in HP (Horizontal Pitch)</param>
+  <param name="desc_unit" type="description"> (Panels are usually defined in units of 2)</param>
   <param name="hp" type="int" min="2" max="64" _gui-text="HP">6</param>
-  <param name="calc" type="description">Set "offset" value for Doepfer-style calculations (.3mm to .4mm), or choose "symmetrical" option for even 7.5mm spacing at edges.</param>
-  <param name="offset" type="float" min="0" max="1" precision="2" _gui-text="Offset">0.36</param>
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.inx:11]: reorder args and more help text

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 2]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
         self.OptionParser.add_option('-o', '--offset', action='store', type='float', dest='offset', default=0.36, help='Amount of material to remove for fitting?')
         self.OptionParser.add_option('-s', '--symmetrical', action='store', type='inkbool', dest='symmetrical', default='False', help='Remove material from both sides?')
         self.OptionParser.add_option('-v', '--oval', action='store', type='inkbool', dest='oval', default='False', help='Oval holes?')
-        self.OptionParser.add_option("-u", "--unit", action="store", type="string", dest="unit", default="mm", help="The unit of the box dimensions")
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py:22]: units weren't in use. changed to another method as units always defined in mm

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 3]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
...
         self.OptionParser.add_option('-c', '--centers', action='store', type='inkbool', dest='centers', default='False', help='Mark centers?')
+        self.OptionParser.add_option('-l', '--lasercut', action='store', type='inkbool', dest='lasercut', default='False', help='Lasercut style?')
+
+    def draw_SVG_Panel(self, (w,h), (x,y), (rx,ry), parent):
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py]: pulled all of these up to the top of the class

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 4]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
...
-        grp = inkex.etree.SubElement(self.current_layer, 'g', grp_attribs)#the group to put everything in
 
+    def effect(self):
 
         hp = self.options.hp
         symmetrical = self.options.symmetrical
         offset = self.options.offset
         oval = self.options.oval
         centers = self.options.centers
+        unitfactor = self.unittouu('1mm') # all our dimensions are in mm
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py]: here's one way to do units

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 5]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
...
-                circ = inkex.etree.SubElement(parent, inkex.addNS('rect','svg'), attribs)
-
-        #parent = self.current_layer
+        # Calculate final width and height of panel
+        pheight = height * unitfactor
+        pwidth =  width * unitfactor
+        
+        # Build top level group to put everything in
+        # Put in in the centre of current view
+        group_transform = 'translate(%s,%s)' % (self.view_center[0]-pwidth/2, self.view_center[1]-pheight/2 )
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py]: transformed so always in center of user's view

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 6]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
...
-                #Top Right
-                draw_SVG_ellipse((rx, ry), (cxr, cyt), grp, start, end)
-
+                # Bottom Right
+                self.draw_SVG_ellipse((rx, ry), (rightH, bottomH), group, 0, end)
+                # Top Right
+                self.draw_SVG_ellipse((rx, ry), (rightH, topH), group, 0, end)
                 # Draw Right-side Centers
                 if centers == True:
-                    #Bottom Right Centers
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py:214]: refactor simplifies this block IMHO

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 7]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
...
+            oval_height = HoleRadius*2*unitfactor
+
+            # Bottom Left
+            self.draw_SVG_square((oval_width,oval_height), (leftH-oval_stretch*unitfactor,bottomH-holeR), (holeR,0), group)
+            # Top Left
+            self.draw_SVG_square((oval_width,oval_height), (leftH-oval_stretch*unitfactor,topH-holeR), (holeR,0), group)
 
             # Draw Left-side Centers
             if centers == True:
-                #Bottom Left Centers
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py:263]: and this one

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 8]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
...
-                #Top Right
-                draw_SVG_square((ovalw,ovalh), (cxr,cyt), (ovals,0), grp)
+                # Bottom Right
+                self.draw_SVG_square((oval_width,oval_height), (rightH-oval_stretch*unitfactor,bottomH-holeR), (holeR,0), group)
+                # Top Right
+                self.draw_SVG_square((oval_width,oval_height), (rightH-oval_stretch*unitfactor,topH-holeR), (holeR,0), group)
 
                 # Draw Left-side Centers
                 if centers == True:
-                #Bottom Right Centers
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py]: and this one

File: EurorackPanelDesigner/INSTALL.txt [hunk 9]
```diff
@@ -1,5 +1,25 @@
 Eurorack Panel Designer
 
-Unarchive and copy files to Inkscape's "extensions" directory (usually inkscape/share/extensions).
+Unarchive and copy files to Inkscape's "extensions" directory.
```
Comment (Neon22) [inline EurorackPanelDesigner/INSTALL.txt:3]: Apparently this is recommended install dir instruction now, as guaranteed to be correct for every user.

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 10]
```diff
@@ -2,15 +2,23 @@
...
+# Adds lasercutting color and refactoring
 
 import sys
 import math
 
 import inkex
 from simplestyle import *
 
+Orange = '#f6921e'
+Blue =   '#0000FF'
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py:19]: I put these colors and styles up here for easy adj

File: EurorackPanelDesigner/EurorackPanelDesigner.py [hunk 11]
```diff
@@ -19,358 +27,237 @@ def __init__(self):
...
+                    # Vertical Line
+                    self.draw_SVG_line( (rightH, bottomH+holeR-gap), (rightH, bottomH-holeR+gap), group)
+                    # Top Right Centers - Horizontal Line
+                    self.draw_SVG_line( (rightH-holeR+gap, topH), (rightH+holeR-gap, topH), group)
+                    # Vertical Line
+                    self.draw_SVG_line( (rightH, topH+holeR-gap), (rightH, topH-holeR+gap), group)
+
+        else: # oval == True
+            # Oval Holes: (a square with rounded corners)
+            oval_size = 5.5  # 3.2mm hole. Oval is 5.5mm across
```
Comment (Neon22) [inline EurorackPanelDesigner/EurorackPanelDesigner.py]: Pretty sure oval dimension can now be adjusted too in this one place.. But not tried it

**Comment by Neon22** [comment]
This now upgraded to inkscape 1.0
included 1U option.
Posted to Inkscape extensions folder.

File: EurorackPanelDesigner/EurorackPanelDesigner.inx (hunk 1 above)
Comment (THX2112) [reply EurorackPanelDesigner/EurorackPanelDesigner.inx:11]: Thanks! Works great!

**Pull request merged**